### PR TITLE
DEV-3519-feat(piscine-go): add solution/test for new go exercise dealapackofcards

### PR DESCRIPTION
**Pull request by tiagocollot**
Created and added solution and test for the new go exercise dealapackofcards

### CODE REVIEW

**Review by jotapero — APPROVED** [review]

File: solutions/dealapackofcards.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package solutions
+
+import "fmt"
+
+func DealAPackOfCards (deck []int) {
+	for _,v := range deck {
+		if (v == 1 && v < 3)  {
```
Comment (eslopfer) [inline solutions/dealapackofcards.go:7]: This condition doesn't make sense.

File: solutions/dealapackofcards.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			fmt.Printf("Player 1: %v, %v, %v", deck[0], deck[1], deck[2])
+			fmt.Println()
+		} else if (v == 3 && v < 6 ){
```
Comment (eslopfer) [inline solutions/dealapackofcards.go:10]: This condition doesn't make sense.

File: solutions/dealapackofcards.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+			fmt.Printf("Player 2: %v, %v, %v", deck[3], deck[4], deck[5])
+			fmt.Println()
+		} else if ( v == 6 && v < 9){
+			fmt.Printf("Player 3: %v, %v, %v", deck[6], deck[7], deck[8])
+			fmt.Println()
+		} else if (v == 9 && v <= 11){
```
Comment (eslopfer) [inline solutions/dealapackofcards.go:16]: This condition doesn't make sense.

File: tests/dealapackofcards_test/main.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	student "student"
+
+	"github.com/01-edu/go-tests/lib/challenge"
+	"github.com/01-edu/go-tests/solutions"
+)
+
+func main() {
+	table := [][]int{{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12},
+									 {1, 2, 3, 4, 5, 17, 7  , 8, 9, 10   , 11 , 12},
```
Comment (eslopfer) [inline tests/dealapackofcards_test/main.go:12]: This wrong indentation is on the source code, it's not a problem of how GitHub renders it.
Also, adding spaces to integers has no meaning.

**Review by eslopfer — CHANGES_REQUESTED** [review]

File: solutions/dealapackofcards.go (hunk 2 above)
Comment (tiagocollot) [reply solutions/dealapackofcards.go]: Thank for your observation. The code is refactored.

File: solutions/dealapackofcards.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+func DealAPackOfCards (deck []int) {
+	for _,v := range deck {
+		if (v == 1 && v < 3)  {
+			fmt.Printf("Player 1: %v, %v, %v", deck[0], deck[1], deck[2])
+			fmt.Println()
+		} else if (v == 3 && v < 6 ){
+			fmt.Printf("Player 2: %v, %v, %v", deck[3], deck[4], deck[5])
+			fmt.Println()
+		} else if ( v == 6 && v < 9){
```
[the remark this answers is not in the document]
Comment (tiagocollot) [reply solutions/dealapackofcards.go]: Thank for your observation. The code is refactored.

File: solutions/dealapackofcards.go (hunk 3 above)
Comment (tiagocollot) [reply solutions/dealapackofcards.go]: Thank for your observation. The code is refactored.

File: tests/dealapackofcards_test/main.go (hunk 4 above)
Comment (tiagocollot) [reply tests/dealapackofcards_test/main.go:12]: The indentation is fixed,  and excess spaces removed.

File: solutions/dealapackofcards.go (hunk 1 above)
Comment (tiagocollot) [reply solutions/dealapackofcards.go]: Thanks for your observation. The code is refactored.

File: tests/dealapackofcards_test/main.go (hunk 4 above)
Comment (tiagocollot) [reply tests/dealapackofcards_test/main.go]: Also removed unnecessary edge cases!

**Review by eslopfer — APPROVED** [review]
LGTM! :rocket:
